src/solvers/petscdmlibmeshimpl.C

```diff
@@ -282,20 +284,19 @@ static PetscErrorCode  DMCreateFieldDecomposition_libMesh(DM dm, PetscInt * len,
     std::map<std::string, unsigned int>  dvarids;
     std::map<unsigned int, std::string>  dvarnames;
     unsigned int                         dvcount = 0;
-    for (std::set<unsigned int>::const_iterator dvit = (*dlm->decomposition)[d].begin(); dvit != (*dlm->decomposition)[d].end(); ++dvit){
-      unsigned int v = *dvit;
+    for (const auto & v : (*dlm->decomposition)[d]) {
       std::string vname = (*dlm->varnames)[v];
       dvarids.insert(std::pair<std::string, unsigned int>(vname,v));
       dvarnames.insert(std::pair<unsigned int,std::string>(v,vname));
       if (!dvcount) dname = vname;
       else   dname += "_" + vname;
       ++dvcount;
       if (!islist) continue;
-      /* Iterate only over this DM's blocks. */
-      for (std::map<std::string, unsigned int>::const_iterator bit = dlm->blockids->begin(); bit != dlm->blockids->end(); ++bit) {
-        unsigned int b = bit->second;
-        MeshBase::const_element_iterator el     = sys->get_mesh().active_local_subdomain_elements_begin(b);
-        MeshBase::const_element_iterator end_el = sys->get_mesh().active_local_subdomain_elements_end(b);
+      // Iterate only over this DM's blocks.
+      for (const auto & pr : *(dlm->blockids)) {
+        MeshBase::const_element_iterator
+          el     = sys->get_mesh().active_local_subdomain_elements_begin(pr.second),
+          end_el = sys->get_mesh().active_local_subdomain_elements_end(pr.second);
         for ( ; el != end_el; ++el) {
```

Comment:
Why not change this to another as_range() use?

Reply:
Must not have matched any of the various regexes I attempted to employ, will update now.